src/python/module/smartredis/util.py

```diff
@@ -111,3 +114,136 @@ def typecheck(arg, name, _type):
     """
     if not isinstance(arg, _type):
         raise TypeError(f"Argument {name} is of type {type(arg)}, not {_type}")
+
+
+
+class DatasetConverter:   
+    
+    def add_metadata_for_xarray(ds,data_names,dim_names,coord_names=None,attr_names=None):
```

Comment:
In other library functions we've used ``dataset`` is input variable name and not ``ds``.  Maybe we should use ``dataset`` in throughout your code to be consistent?